rfcs/0001-masonry-backend.md

```diff
@@ -0,0 +1,166 @@
+# Feature Name: masonry-backend
+
+## Summary
+
+Use the codebase of the crate [masonry-rs](https://github.com/PoignardAzur/masonry-rs) as the new backend for Xilem.
+
+## Motivation
+
+Masonry and Xilem were both forked from Druid, and currently share a lot of code and design.
+
+Masonry was written as an exploration into the concept of a GUI backend, with a Widget tree that could serve as a the retained component of a reactive architecture.
+
+Xilem was written as an exploration into reactive GUI, and thus more effort was spent on its frontend and on how add developers would describe their interfaces.
+
+As a result, Xilem's native backend is in a poor state:
+
+- There is [code commented out](https://github.com/linebender/xilem/blob/ea45b9f8c14e3708f0fcbe0a0e1c760f59146323/src/widget/widget.rs#L113-L120).
+- There are [*entire modules* commented out](https://github.com/linebender/xilem/blob/ea45b9f8c14e3708f0fcbe0a0e1c760f59146323/src/widget/mod.rs#L19-L20).
+- There is [documentation referring to items from Druid that no longer exist](https://github.com/linebender/xilem/blob/ea45b9f8c14e3708f0fcbe0a0e1c760f59146323/src/widget/widget.rs#L51-L71).
+- There are [TODOs without an associated issue](https://github.com/linebender/xilem/blob/ea45b9f8c14e3708f0fcbe0a0e1c760f59146323/src/widget/core.rs#L569C5-L569C66).
+
+While Masonry's codebase hasn't been updated in a while, I made sure to address the above problems before I stepped away from it. I would also argue it has a tighter vision and more helpful documentation, thought that is more subjective.
+
+Masonry also comes with some built-in perks, like powerful unit tests and a structured widget graph.
+
+For these reasons, I think a new backend should be added to Xilem using either the Masonry crate or one forked from it. (We will refer to that new backend as "Masonry" in the rest of this RFC, although it might have a different name in the end.)
+
+
+## User-facing explanation
+
+### Xilem architecture
+
+Xilem is roughly separated into two layers:
+
+- **The reactive layer** handles the app logic and centralized state. Your application's code will call functions from this layer.
+- **The platform layer** handles the document state, depending on the platform.
+
+The reactive layer defines the `View` trait as well as other items for sequences, type-erasure and memoization. A `View` represent a transient node or tree of graphical items, which is return by a declarative function we usually call a "component". An example of component could be:
```

Comment:
```suggestion
The reactive layer defines the `View` trait as well as other items such as sequences of views, type-erasure and memoization. A `View` represents a transient description of one or multiple graphical items, which is usually returned by a declarative function that is often called a "component". An example of component could be:
```

Reply:
Not really resolved, surely this is a little bit of nitpicking... but I don't see why the suggestion is not taken (as it should be a little bit more precise). And yes, I don't think the whole explanation of Xilems reactive architecture is necessary in this RFC (but since you've already written it, we can at least make it more accurate).

Reply:
> I don't see why the suggestion is not taken

The current text in the RFC:

> A `View` represents a transient description of one or multiple graphical items, which is usually returned by a declarative function that is often called a "component".

Is taken directly from your wording. I didn't keep the "such as sequences of views, type-erasure and memoization" part because the sentence "reactive layer defines [...] items such as [...] type-erasure and memoization" doesn't really parse right.